src/components/FilterButton/FilterButton.js

```diff
@@ -0,0 +1,91 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Button from 'react-spatial/components/Button';
+import filterActive from '../../images/FilterButton/filterActive.png';
+import filterInactive from '../../images/FilterButton/filterInactive.png';
+
+import TrackerLayer from '../../layers/TrackerLayer';
+import TrajservLayer from '../../layers/TrajservLayer';
+
+const propTypes = {
+  /**
+   * CSS class of the filter button.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Title.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Line info route identifer.
+   */
+  routeIdentifier: PropTypes.string.isRequired,
+
+  /**
+   * Trackerlayer.
+   */
+  trackerLayer: PropTypes.instanceOf(TrackerLayer).isRequired,
+};
+
+const defaultProps = {
+  className: 'rt-control-button rt-route-filter',
+  title: 'Filter',
+};
+
+/**
+ * Displaying all stops of a line, and their informations.
+ */
+class FilterButton extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filterActivated: false,
+    };
+  }
+
+  toggleFilter(routeIdentifier, trackerLayer) {
```

Comment:
`trackerLayer` is a prop. For readability let's not pass around props as function parameters.